qunit-composite.js

```diff
@@ -38,6 +38,36 @@ function runSuite( suite ) {
 		path = suite;
 	}
 
+	// Parse this iframe's query string
+	query = parseQuery( path );
+
+	// If queryTop or query is not empty...
+	if ( queryTop || query ) {
+		// Replace this iframe's query keys with matching keys from top frame
+		var search = $.extend( query, queryTop ),
+			urlParts = path.split( "?" ),
+			pathParts = path.split( "#" ),
+			hash = pathParts[ 1 ],
+			searchPairs = [], pathQuery;
+
+		// If there is anything after "?",
+		// take only that's what's before it for now
+		if ( urlParts.length > 1 || pathParts.length > 1 ) {
+			path = pathParts[ 0 ].split( "?" )[ 0 ];
+		}
```

Comment:
Why so much manual parsing? You can just create an anchor so you get a `Location` object to work with.